product_list_price_from_pricelist/models/product_pricelist.py

```diff
@@ -0,0 +1,142 @@
+from odoo import api, models
+from odoo.tools import config
+
+
+class ProductPricelist(models.Model):
+    _inherit = "product.pricelist"
+
+    def _update_product_price_from_pricelist(self, pricelist_items=None):
+        self.ensure_one()
+        all_templates = self.env["product.template"]
+        if not self._can_update_product_price():
+            return False
+        pricelist_items = self.item_ids if not pricelist_items else pricelist_items
+        for item in pricelist_items:
+            templates = item._get_all_templates_from_pricelist_item()
+            all_templates |= templates
+        if all_templates:
+            pricelist_data = self._compute_price_rule(all_templates, 1)
+            for template in all_templates:
+                new_price, suitable_rule = pricelist_data[template.id]
+                if suitable_rule and new_price != template.list_price:
+                    template.write({"list_price": new_price})
+        return True
+
+    def _can_update_product_price(self):
```

Comment:
Instead of having a method that gets the domain and then do a bool operation, do a method that returns the domain, and apply it directly on `_get_all_templates_from_pricelist_item` for getting the applied items, returning empty if not applicable. This way, the logic is more direct.